java/common/src/main/scala/org/typelevel/otel4s/java/TextMapPropagatorImpl.scala

```diff
@@ -39,4 +39,16 @@ private[java] class TextMapPropagatorImpl[F[_]: Sync](
     Sync[F].delay(
       jPropagator.inject(toJContext(ctx), carrier, fromTextMapSetter)
     )
+
+  def injected[A](ctx: Vault, carrier: A)(implicit
+      injector: TextMapInjector[A]
+  ): A = {
+    val builder = injector.toBuilder(carrier)
+    jPropagator.inject(
+      toJContext(ctx),
+      builder,
+      fromTextMapSetter(injector.textMapSetter)
+    )
+    injector.toCarrier(builder)
```

Comment:
Good: the mutation is hidden from the outside.

Bad: allocates a builder and carrier even if no fields are injected.  Can we ask `jPropagator` whether it's actually going to inject anything?  We could do a decorator and lazily create the builder.

Awkward: this implementation needs `Sync` just for the `inject`, which I think would be less used in Scala than this variant!

Reply:
you could always check if the vault is empty first. idk if that covers *all* cases, but it could optimise some